### PR TITLE
Updating the readme

### PR DESCRIPTION
**Pull request by peterBrxwn**

### CODE REVIEW

**Comment by peterBrxwn** [comment]
[[Flutter] Prepare a better readme file (10.01)](https://trello.com/c/WMuI9IQg/1174-flutter-prepare-a-better-readme-file-1001)

File: README.md [hunk 1]
```diff
@@ -29,6 +29,144 @@ This SDK supports the following features:
 * [Performance Monitoring](https://support.count.ly/hc/en-us/articles/4734457847705-Performance)
 * [Feedback Widgets](https://support.count.ly/hc/en-us/articles/4652903481753-Feedback-Surveys-NPS-and-Ratings-)
 
+## Installation
+
+In the `dependencies:` section of your `pubspec.yaml`, add the following line:
+
+```yaml
+dependencies:
+  countly_flutter: <latest_version>
```
Comment (arifBurakDemiray) [inline README.md:38]: maybe using version would be better, but it should be maintanied

File: README.md [hunk 2]
```diff
@@ -29,6 +29,144 @@ This SDK supports the following features:
...
+    );
+  }
+}
+```
+
+## Enabling Automatic Crash Handling
+
+```dart
+final CountlyConfig config = CountlyConfig(SERVER_URL, APP_KEY)
+  ..enableCrashReporting()
```
Comment (arifBurakDemiray) [inline README.md:93]: there is two dot. Is it a flutter thing

File: README.md [hunk 3]
```diff
@@ -29,6 +29,144 @@ This SDK supports the following features:
...
+await Countly.changeDeviceId('123456', true);
+
+// without merge
+await Countly.changeDeviceId('123456', false);
+```
+
+## Get Device ID Type
+
+```dart
+final DeviceIdType? deviceIdtype = await Countly.getDeviceIDType();
```
Comment (arifBurakDemiray) [inline README.md:132]: maybe also mentioning current device types would be better

File: README.md [hunk 4]
```diff
@@ -29,6 +29,144 @@ This SDK supports the following features:
...
+```
+
+## User Profile
+
```
Comment (arifBurakDemiray) [inline README.md:136]: maybe an example about custom ops

File: scripts/no-push-files/README.md [hunk 5]
```diff
@@ -28,6 +28,144 @@ This SDK supports the following features:
 * [Performance Monitoring](https://support.count.ly/hc/en-us/articles/4734457847705-Performance)
 * [Feedback Widgets](https://support.count.ly/hc/en-us/articles/4652903481753-Feedback-Surveys-NPS-and-Ratings-)
```
Comment (arifBurakDemiray) [inline scripts/no-push-files/README.md:30]: This file is duplicate and should be removed I guess

Comment (arifBurakDemiray) [reply scripts/no-push-files/README.md:30]: Oh I got it is a bizarre implementation

**Review by arifBurakDemiray — CHANGES_REQUESTED** [review]
Countly Flex is missing in intro
https://github.com/Countly/countly-sdk-java/blob/8e35f43383e4b532df91c66d526de47233422cf1/README.md?plain=1#L5

Looking for county server line should use server repo link
https://github.com/Countly/countly-sdk-android/blob/1067d2968bc10a683a51a40ff7a374cb814288fb/README.md?plain=1#L15

image
These should be in new line

**Review by turtledreams — CHANGES_REQUESTED** [review]
as you are at it can you change the codacy badge and title of the np readme

File: README.md (hunk 1 above)
Comment (peterBrxwn) [reply README.md:38]: Yeah, we could do that but the package we are trying to emulate did not.

File: README.md (hunk 2 above)
Comment (peterBrxwn) [reply README.md:93]: yes

File: README.md (hunk 1 above)
Comment (peterBrxwn) [reply README.md:38]: https://pub.dev/packages/font_awesome_flutter#installation

File: README.md (hunk 3 above)
Comment (peterBrxwn) [reply README.md:132]: added

**Review by ArtursKadikis — CHANGES_REQUESTED** [review]
rework it with more context

File: scripts/no-push-files/README.md [hunk 6]
```diff
@@ -28,6 +32,172 @@ This SDK supports the following features:
...
+}
+
+class _MyAppState extends State<MyApp> {
+  @override
+  void initState() {
+    super.initState();
+
+    Countly.isInitialized().then((bool isInitialized) {
+      if (!isInitialized) {
+        final CountlyConfig config = CountlyConfig(SERVER_URL, APP_KEY)..setLoggingEnabled(true);
```
Comment (turtledreams) [inline scripts/no-push-files/README.md:69]: comment here what we are doing

File: scripts/no-push-files/README.md [hunk 7]
```diff
@@ -28,6 +32,172 @@ This SDK supports the following features:
...
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(title: Text('Countly Example App')),
+      body: Center(
+        child: TextButton(
+          onPressed: () {
+            Countly.recordEvent({'key': 'Basic Event', 'count': 1});
```
Comment (turtledreams) [inline scripts/no-push-files/README.md:82]: comment here what we are doing

File: scripts/no-push-files/README.md [hunk 8]
```diff
@@ -28,6 +32,172 @@ This SDK supports the following features:
...
+          child: Text('Record Event'),
+        ),
+      ),
+    );
+  }
+}
+```
+
+## Enabling Automatic Crash Handling
+
```
Comment (turtledreams) [inline scripts/no-push-files/README.md:93]: add a small text explaining what we are doing

File: scripts/no-push-files/README.md [hunk 9]
```diff
@@ -28,6 +32,172 @@ This SDK supports the following features:
...
+```dart
+final CountlyConfig config = CountlyConfig(SERVER_URL, APP_KEY)
+  ..enableCrashReporting()
+Countly.initWithConfig(config);
+```
+
+## Reporting Exceptions manually
+
```
Comment (turtledreams) [inline scripts/no-push-files/README.md:101]: add a small text explaining what we are doing
manually => Manually

File: scripts/no-push-files/README.md [hunk 10]
```diff
@@ -28,6 +32,172 @@ This SDK supports the following features:
...
+```dart
+Countly.logException('This is a manually created exception', true, null);
+```
+
+## Recording Events
+
```
Comment (turtledreams) [inline scripts/no-push-files/README.md:107]: add a small text explaining what we are doing

File: scripts/no-push-files/README.md [hunk 11]
```diff
@@ -28,6 +32,172 @@ This SDK supports the following features:
...
+```dart
+Countly.recordEvent({'key': 'Basic Event', 'count': 1});
+```
+
+## Recording Views
+
```
Comment (turtledreams) [inline scripts/no-push-files/README.md:113]: add a small text explaining what we are doing

File: scripts/no-push-files/README.md [hunk 12]
```diff
@@ -28,6 +32,172 @@ This SDK supports the following features:
...
+```dart
+// start recording view
+await Countly.instance.views.startView('HomePage');
+
+// stop recording view
+await Countly.instance.views.stopViewWithName('HomePage');
+```
+
+## Change Device ID
+
```
Comment (turtledreams) [inline scripts/no-push-files/README.md:123]: add a small text explaining what we are doing

File: scripts/no-push-files/README.md [hunk 13]
```diff
@@ -28,6 +32,172 @@ This SDK supports the following features:
...
+```dart
+// with merge
+await Countly.changeDeviceId('123456', true);
+
+// without merge
+await Countly.changeDeviceId('123456', false);
+```
+
+## Get Device ID Type
+
```
Comment (turtledreams) [inline scripts/no-push-files/README.md:133]: add a small text explaining what we are doing

File: scripts/no-push-files/README.md [hunk 14]
```diff
@@ -28,6 +32,172 @@ This SDK supports the following features:
...
+```dart
+final DeviceIdType? deviceIdtype = await Countly.getDeviceIDType();
+// DeviceIdType: DEVELOPER_SUPPLIED, SDK_GENERATED, TEMPORARY_ID
+```
+
+## User Profile
+
```
Comment (turtledreams) [inline scripts/no-push-files/README.md:140]: add a small text explaining what we are doing

File: scripts/no-push-files/README.md [hunk 15]
```diff
@@ -28,6 +32,172 @@ This SDK supports the following features:
...
+// Send/Save provided values to server
+Countly.instance.userProfile.save();
+
+// Clear queued operations / modifications
+Countly.instance.userProfile.clear();
+```
+
+## Consent
+
```
Comment (turtledreams) [inline scripts/no-push-files/README.md:186]: add a small text explaining what we are doing

File: scripts/no-push-files/README.md [hunk 16]
```diff
@@ -1,8 +1,8 @@
 [![Codacy Badge](https://app.codacy.com/project/badge/Grade/7ab95afb3da8421ab499cc921e1381ac)](https://www.codacy.com/gh/Countly/countly-sdk-flutter-bridge/dashboard?utm_source=github.com&amp;utm_medium=referral&amp;utm_content=Countly/countly-sdk-flutter-bridge&amp;utm_campaign=Badge_Grade)
 
-# Countly Flutter SDK
+# Countly Flutter SDK - No Push
 
-This repository contains the Countly Flutter SDK, which can be integrated into mobile Flutter applications. The Countly Flutter SDK is intended to be used with [Countly Lite](https://github.com/Countly/countly-server) or [Countly Enterprise](https://count.ly/product).
+This repository contains the Countly Flutter SDK, which can be integrated into mobile Flutter applications. The Countly Flutter SDK is intended to be used with [Countly Lite](https://countly.com/lite), [Countly Flex](https://countly.com/flex), [Countly Enterprise](https://countly.com/enterprise).
 
 ## What is Countly?
 [Countly](https://count.ly) is a product analytics solution and innovation enabler that helps teams track product performance and customer journey and behavior across [mobile](https://count.ly/mobile-analytics), [web](https://count.ly/web-analytics),
```
Comment (turtledreams) [inline scripts/no-push-files/README.md:8]: all count.ly s should be countly.com

File: README.md [hunk 17]
```diff
@@ -1,8 +1,8 @@
...
+[![Codacy Badge](https://app.codacy.com/project/badge/Grade/7ab95afb3da8421ab499cc921e1381ac)](https://app.codacy.com/gh/Countly/countly-sdk-flutter-bridge/dashboard?utm_source=gh&utm_medium=referral&utm_content=&utm_campaign=Badge_grade)
 
 # Countly Flutter SDK
 
-This repository contains the Countly Flutter SDK, which can be integrated into mobile Flutter applications. The Countly Flutter SDK is intended to be used with [Countly Lite](https://github.com/Countly/countly-server) or [Countly Enterprise](https://count.ly/product).
+This repository contains the Countly Flutter SDK, which can be integrated into mobile Flutter applications. The Countly Flutter SDK is intended to be used with [Countly Lite](https://countly.com/lite), [Countly Flex](https://countly.com/flex), [Countly Enterprise](https://countly.com/enterprise).
 
 ## What is Countly?
 [Countly](https://count.ly) is a product analytics solution and innovation enabler that helps teams track product performance and customer journey and behavior across [mobile](https://count.ly/mobile-analytics), [web](https://count.ly/web-analytics),
```
Comment (turtledreams) [inline README.md:8]: all count.ly => countly.com

File: README.md [hunk 18]
```diff
@@ -29,6 +33,172 @@ This SDK supports the following features:
...
+}
+
+class _MyAppState extends State<MyApp> {
+  @override
+  void initState() {
+    super.initState();
+
+    Countly.isInitialized().then((bool isInitialized) {
+      if (!isInitialized) {
+        final CountlyConfig config = CountlyConfig(SERVER_URL, APP_KEY)..setLoggingEnabled(true);
```
Comment (turtledreams) [inline README.md:70]: add a small comment explaining what we are doing

File: README.md [hunk 19]
```diff
@@ -29,6 +33,172 @@ This SDK supports the following features:
...
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(title: Text('Countly Example App')),
+      body: Center(
+        child: TextButton(
+          onPressed: () {
+            Countly.recordEvent({'key': 'Basic Event', 'count': 1});
```
Comment (turtledreams) [inline README.md:83]: add a small comment explaining what we are doing

File: README.md [hunk 20]
```diff
@@ -29,6 +33,172 @@ This SDK supports the following features:
...
+          child: Text('Record Event'),
+        ),
+      ),
+    );
+  }
+}
+```
+
+## Enabling Automatic Crash Handling
+
```
Comment (turtledreams) [inline README.md:94]: add a small text explaining what we are doing

File: README.md [hunk 21]
```diff
@@ -29,6 +33,172 @@ This SDK supports the following features:
...
+```dart
+final CountlyConfig config = CountlyConfig(SERVER_URL, APP_KEY)
+  ..enableCrashReporting()
+Countly.initWithConfig(config);
+```
+
+## Reporting Exceptions manually
+
```
Comment (turtledreams) [inline README.md:102]: add a small text explaining what we are doing
m => M

File: README.md [hunk 22]
```diff
@@ -29,6 +33,172 @@ This SDK supports the following features:
...
+```dart
+Countly.logException('This is a manually created exception', true, null);
+```
+
+## Recording Events
+
```
Comment (turtledreams) [inline README.md:108]: add a small text explaining what we are doing

File: README.md [hunk 23]
```diff
@@ -29,6 +33,172 @@ This SDK supports the following features:
...
+```dart
+Countly.recordEvent({'key': 'Basic Event', 'count': 1});
+```
+
+## Recording Views
+
```
Comment (turtledreams) [inline README.md:114]: add a small text explaining what we are doing

File: README.md [hunk 24]
```diff
@@ -29,6 +33,172 @@ This SDK supports the following features:
...
+```dart
+// start recording view
+await Countly.instance.views.startView('HomePage');
+
+// stop recording view
+await Countly.instance.views.stopViewWithName('HomePage');
+```
+
+## Change Device ID
+
```
Comment (turtledreams) [inline README.md:124]: add a small text explaining what we are doing

File: README.md [hunk 25]
```diff
@@ -29,6 +33,172 @@ This SDK supports the following features:
...
+```dart
+// with merge
+await Countly.changeDeviceId('123456', true);
+
+// without merge
+await Countly.changeDeviceId('123456', false);
+```
+
+## Get Device ID Type
+
```
Comment (turtledreams) [inline README.md:134]: add a small text explaining what we are doing

File: README.md [hunk 26]
```diff
@@ -29,6 +33,172 @@ This SDK supports the following features:
...
+```dart
+final DeviceIdType? deviceIdtype = await Countly.getDeviceIDType();
+// DeviceIdType: DEVELOPER_SUPPLIED, SDK_GENERATED, TEMPORARY_ID
+```
+
+## User Profile
+
```
Comment (turtledreams) [inline README.md:141]: add a small text explaining what we are doing

File: README.md [hunk 27]
```diff
@@ -29,6 +33,172 @@ This SDK supports the following features:
...
+// Send/Save provided values to server
+Countly.instance.userProfile.save();
+
+// Clear queued operations / modifications
+Countly.instance.userProfile.clear();
+```
+
+## Consent
+
```
Comment (turtledreams) [inline README.md:187]: add a small text explaining what we are doing

**Review by turtledreams — CHANGES_REQUESTED** [review]

File: scripts/no-push-files/pubspec.yaml [hunk 28]
```diff
@@ -1,7 +1,11 @@
 name: countly_flutter_np
 description: Countly is an innovative, real-time, open source mobile analytics. This plugin is a flavour of the SDK without push notifications.
-version: 23.12.1
+version: 23.12.0
```
Comment (ijunaid) [inline scripts/no-push-files/pubspec.yaml:3]: Please remove this line, we also have version on line # 8

File: scripts/no-push-files/README.md [hunk 29]
```diff
@@ -28,26 +32,239 @@ This SDK supports the following features:
 * [Performance Monitoring](https://support.count.ly/hc/en-us/articles/4734457847705-Performance)
 * [Feedback Widgets](https://support.count.ly/hc/en-us/articles/4652903481753-Feedback-Surveys-NPS-and-Ratings-)
 
+## Installation
+
+In the `dependencies:` section of your `pubspec.yaml`, add the following line:
+
+```yaml
+dependencies:
+  countly_flutter: <latest_version>
```
Comment (ijunaid) [inline scripts/no-push-files/README.md:41]: is it should be 'countly_flutter_np'?

File: scripts/no-push-files/README.md [hunk 30]
```diff
@@ -28,26 +32,239 @@ This SDK supports the following features:
...
+```
+
+## Usage
+
+```dart
+import 'package:countly_flutter/countly_flutter.dart';
```
Comment (ijunaid) [inline scripts/no-push-files/README.md:47]: Please update import with _np

**Review by ijunaid — CHANGES_REQUESTED** [review]

**Pull request merged**